src/barretenberg_structures.rs

```diff
@@ -302,6 +302,34 @@ impl HashToFieldConstraint {
         buffer
     }
 }
+
+#[derive(Clone, Hash, Debug)]
+pub(crate) struct HashConstraint {
+    pub(crate) inputs: Vec<(i32, i32)>,
+    pub(crate) result: [i32; 32],
+}
+
+impl HashConstraint {
```

Comment:
This name is too generic as we have many hash constraints

Reply:
Yes, it is on purpose, I plan to use the same structure for sha256 and blake2s.

Reply:
We can rename it when this struct has been generalised in that case.